### ISSUE TITLE
phasez returns negative phase

### ISSUE THREAD

**Issue by Firionus** [JuliaDSP/DSP.jl#310]
For example:
```julia
julia> angle.(freqz(digitalfilter(Lowpass(.5), Butterworth(2))))
250-element Array{Float64,1}:
  0.0
 -0.008921688379769978
 -0.017844797030280853
  ⋮
 -3.123747856559511
 -3.132670965210026
 -3.141592653589793

julia> phasez(digitalfilter(Lowpass(.5), Butterworth(2)))
250-element Array{Float64,1}:
 -0.0
  0.008921688379769978
  0.017844797030280853
  ⋮
  3.123747856559511
  3.132670965210026
  3.141592653589793
```
In this case `phasez` gives back phase that rises with frequency which is associated with negative group delay that should not appear with any causal filter. 

The function definition states:
```julia
function phasez(filter::FilterCoefficients, w = range(0, stop=π, length=250))
    h = freqz(filter, w)
    unwrap(-atan.(imag(h), real(h)); dims=ndims(h))
end
```
and I'm not sure why that minus is before the tangent. Is there any reason for this or should I try to put in my first ever PR with an easy fix?

**Comment by galenlynch** [comment]
I can't think of a reason it should be that way. @mpastell was the original author of that code, and @jfsantos reviewed it in PR #93. Do you guys remember if the minus sign was intentional? If not it would be great if you could go ahead and make a PR to fix it.

**Comment by Firionus** [comment]
I tried changing the sign, which made the test for phasez fail that previously succeeded. 

I then reversed the polynomial order in the test, because according to MATLAB docs freqz takes in the coefficents from low to high order, whereas PolymialRatio takes its coeffcients from high to low order. 

With this "correction" phasez passed the test, but impz and stepz broke. So this issue seems to be an extension of #135, which was analyzed in https://github.com/JuliaDSP/DSP.jl/issues/118#issuecomment-301642700.
